utils/s2n_mem.h

```diff
@@ -26,6 +26,7 @@ int s2n_mem_cleanup(void);
 int s2n_alloc(struct s2n_blob *b, uint32_t size);
 int s2n_realloc(struct s2n_blob *b, uint32_t size);
 int s2n_free(struct s2n_blob *b);
+int s2n_free_without_wipe(struct s2n_blob *b);
 int s2n_blob_zeroize_free(struct s2n_blob *b);
```

Comment:
Agreed

Reply:
So it looks like there is a difference in functions.

`s2n_blob_zeroize_free` _always_ zeroes the memory. 

https://github.com/aws/s2n-tls/blob/120920817625c51a983cfced439a501673d143d6/utils/s2n_mem.c#L300-L303

`s2n_blob_free` will fail if you try and free a non-growable blob:

https://github.com/aws/s2n-tls/blob/120920817625c51a983cfced439a501673d143d6/utils/s2n_mem.c#L285

Reply:
Oh no, that is an annoyingly subtle distinction xD So same behavior on success, but one ignores blobs that can't be freed and the other errors. I added half those references, and I certainly didn't do it because I was relying on that difference. I did it because the memory being freed was sensitive and definitely needed to be zeroed.

Reply:
https://github.com/aws/s2n-tls/issues/3475